// file-droppa/Directives/FileList.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var FileStore_service_1 = require("../Services/FileStore.service");
var FileList = (function () {
    function FileList(filesStore) {
        this.filesStore = filesStore;
    }
    FileList.prototype.removeFile = function (iFile) {
        this.filesStore.removeFiles(iFile);
    };
    return FileList;
}());
FileList.decorators = [
    { type: core_1.Component, args: [{
                selector: 'fileList, [fileList]',
                styles: ["\n        .file-list {\n            width: 430px;\n            margin-bottom: 5px;\n            display: flex;\n            flex-flow: wrap;\n            justify-content: flex-start;\n         }\n    "],
                template: "\n        <div class=\"file-list\">\n            <fileItem *ngFor=\"let file of filesStore.iFiles\"\n                [file]=\"file.File\"\n                [percentage]=\"file.percentage\"\n                [loadingSuccessful]=\"file.loadingSuccessful\"\n                [responseMessage]=\"file.responseMessage\"\n                (removeFile)=\"removeFile(file)\">\n            </fileItem>\n        </div>\n    "
            },] },
];
/** @nocollapse */
FileList.ctorParameters = function () { return [
    { type: FileStore_service_1.FilesStore, },
]; };
exports.FileList = FileList;
